src/dto/signUpDTO.ts

```diff
@@ -3,8 +3,11 @@ import { BaseUserDTO } from './baseUserDTO';
 
 export class SignUpDTO extends BaseUserDTO {
   @IsString()
-  firstName?: string;
+  firstName!: string;
```

Comment:
Are we sure that @isString() is not allowing nulls? We can check the docs https://github.com/typestack/class-validator

Reply:
Same applies for `gender` and `lastName`

Reply:
Added isNotEmpty in new commit